### ISSUE TITLE
virt_text overlay does not scroll horizontally

### ISSUE THREAD

**Issue by lukas-reineke** [neovim/neovim#14050]
- `nvim --version`: NVIM v0.5.0-dev+1116-gbda12927b

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
:set nowrap
5000ik<ESC>
:lua ns_id = vim.api.nvim_create_namespace("foo")
:lua vim.api.nvim_buf_set_extmark(0, ns_id, 0, 1, { virt_text={{'gg', 'String'}}, virt_text_pos='overlay'})
$
```

### Actual behaviour

When scrolling right, the overlay virtual text stays in the same place.

### Expected behaviour

It should scroll with the rest of the text

**Comment by lukas-reineke** [comment]
cc @bfredl

**Comment by bfredl** [comment]
hmm they do scroll for me, except when getting to the edge they stay at the left margin rather than disappear or get left-truncated. For the use case of intend guides we probably want to implement the latter as an option though.

**Comment by lukas-reineke** [comment]
> they do scroll for me, except when getting to the edge they stay at the left margin rather than disappear or get left-truncated

You are right, I only tested with the first column, so I didn't notice that.

An option would be nice :+1:

**Comment by bphilly96** [comment]
I see that #14065 added `virt_text_hide` and referenced this issue however when I use `virt_text_hide=true` the virtual text doesn't appear at all. Is this a bug or am I missing something?

**Comment by WhyNotHugo** [comment]
This has been affecting lsp_lines for a while: https://todo.sr.ht/~whynothugo/lsp_lines.nvim/12

When scrolling to the right, diagnostics shift to the right. The most annoying part of this is being unable to read long text by scrolling to the right.

**Comment by zeertzjq** [comment]
It's strange that `virt_text_hide` serves two purposes at the same time:
1. Hide the overlay virtual text when the extmark is in the text hidden by 'nowrap'
2. Hide the overlay virtual text when the extmark is in Visual selection

And it's not possible only choose one of them.
